test_plans/is_device_copyable.asciidoc

```diff
@@ -0,0 +1,81 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for is_device_copyable
+
+This is a test plan for `is_device_copyable` type trait described in https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_is_device_copyable_type_trait[SYCL 2020 sections 4.12.3. is_device_copyable type trait]
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+=== Types coverage
+
+All of the tests described below are performed using `typename T` which satisfies the https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec::device.copyable[requriments] of a device copyable type:
+
+* The application defines the trait `is_device_copyable_v<T>` to `true`;
+* Type `T` has at least one eligible copy constructor, move constructor, copy assignment operator, or move assignment operator;
+* Each eligible copy constructor, move constructor, copy assignment operator, and move assignment operator is public;
+* When doing an inter-device transfer of an object of type `T`, the effect of each eligible copy constructor, move constructor, copy assignment operator, and move assignment operator is the same as a bitwise copy of the object;
+* Type `T` has a public non-deleted destructor;
+* The destructor has no effect when executed on the device.
+
+Define a new class of type `T` with public default constructor, destructor, copy constructor, copy assignment operator and several public members:
+```
+class T {
+  public:
+    T() = default;
+    ~T() = default;
+
+    /* -- common interface members -- */
+
+    int a;
+    float b;
+    char c;
+};
+```
+
+Given that `is_device_copyable<T>::value` can be either `true` or `false`, `integral_constant_t` will be used instead of `std::integral_constant<bool, true>` and `std::integral_constant<bool, false>`.
```

Comment:
I don't understand this statement.  Doesn't your test program set `is_device_copyable<t>` to true?  Therefore, the value can only be `true`.

Reply:
I tried to add specialization after requirements tests using inheritance, but You are right, I will rewrite it.